vassal-app/src/main/java/VASSAL/counters/Deck.java

```diff
@@ -1668,4 +1669,67 @@ protected void repaintMap() {
       map.repaint();
     }
   }
+
+  /**
+   * Updates the properties of the Deck in the current open game based on the Deck definition stored in the module (stored in DrawPile.dummy)
+   */
+  public void Refresh() {
+    // Locate the Drawpile in the module that matches the Deck
+    // Update properties
+    for (final DrawPile drawPile : gameModule.getAllDescendantComponentsOf(DrawPile.class)) {
+      if (getDeckName().equals(drawPile.getAttributeValueString(SetupStack.NAME))) {
```

Comment:
You can just call `DrawPile.findDrawPile(name)` to find a given named DrawPile.

Reply:
For that matter why not make the DrawPile a parameter e.g. Refresh(DrawPile drawPile) if at the time you're calling this you already know the drawpile in question?